pennylane/devices/qubit/sampling.py

```diff
@@ -269,6 +280,7 @@ def _process_single_shot(samples):
         return tuple(processed)
 
     try:
+        prng_key, _ = jax_random_split(prng_key)
```

Comment:
If we split right before calling this in `measure_with_samples`, do we need to split the key again?

Reply:
You're correct, it is not necessary here. I think we could follow either of two extremes: split only when necessary, or split prior to passing to any function. The former is cleaner. The latter is a bit cumbersome, but then we need not think too much about which key ends up where. If we go with the former I could remove a couple other split in the current PR.